torch/nn/functional.py

```diff
@@ -1909,10 +1909,13 @@ def poisson_nll_loss(input, target, log_input=True, full=False, size_average=Non
         loss[mask] += (target * torch.log(target) - target + 0.5 * torch.log(2 * math.pi * target))[mask]
     if reduction == 'none':
         ret = loss
-    if reduction == 'mean':
+    elif reduction == 'mean':
         ret = torch.mean(loss)
-    else:
+    elif reduction == 'sum':
         ret = torch.sum(loss)
+    else:
+        ret = input
```

Comment:
nit: We don't need this anymore, right?

Reply:
I think JIT will throw up an error, because we wouldn't be true to the type annotation.

Reply:
Interesting! That's fine then, thanks for figuring that out